### ISSUE TITLE
batch_normalization error

### ISSUE THREAD

**Issue by longchr123** [nex3z/tflite-mnist-android#5]
thank you for sharing your code,have you tried this:
```
def batch_norm_lite(x, train=True, bn_decay=0.5,epsilon = 0.001,name='bn'):
    is_training = tf.convert_to_tensor(train,dtype='bool',name='is_training')
    x_shape = x.get_shape()
    params_shape = x_shape[-1:]

    axis = list(range(len(x_shape) - 1))

    beta = tf.get_variable(name+'_beta', params_shape, initializer=tf.zeros_initializer())
    gamma = tf.get_variable(name+'_gamma', params_shape, initializer=tf.ones_initializer())

    moving_mean = tf.get_variable(name+'_moving_mean', params_shape, initializer=tf.zeros_initializer(), trainable=False)
    moving_variance = tf.get_variable(name+'_moving_variance', params_shape, initializer=tf.ones_initializer(), trainable=False)

    # These ops will only be preformed when training.
    mean, variance = tf.nn.moments(x, axis)
    update_moving_mean = moving_averages.assign_moving_average(moving_mean, mean, bn_decay)
    update_moving_variance = moving_averages.assign_moving_average(moving_variance, variance, bn_decay)
    tf.add_to_collection(name+'_update_moving_mean', update_moving_mean)
    tf.add_to_collection(name+'_update_moving_variance', update_moving_variance)


    mean, variance = control_flow_ops.cond(
        is_training, lambda: (mean, variance),
        lambda: (moving_mean, moving_variance))

    return tf.nn.batch_normalization(x, mean, variance, beta, gamma, epsilon,name=name)

def inference(input_tensor, regularizer=None):
    with tf.variable_scope("layer_1_conv"):
        conv_1_weight = get_weight([CONV_1_SIZE, CONV_1_SIZE, IMAGE_CHANNEL_NUM, CONV_1_DEPTH])
        conv_1_bias = get_bias([CONV_1_DEPTH])
        conv_1 = conv2d(input_tensor, conv_1_weight, stride=1)
        bn = batch_norm_lite(tf.nn.bias_add(conv_1, conv_1_bias))
        conv_1_activation = tf.nn.relu(bn)

```

add a batch_normalization layer before relu. Because this is so common, this code cannot be turned into tflite,how to solve this problem?

**Comment by nex3z** [comment]
I've also tried batch normalization with TfLite but received error about "supported operators" when converting the model using toco. I think it's because TensorFlow Lite only supports a subset of full fledged TensorFlow operations as documented [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/tf_ops_compatibility.md).

Perhaps you can try [TensorFlow Mobile](https://www.tensorflow.org/mobile/mobile_intro), which supports more operations. You can find their differences [here](https://www.tensorflow.org/mobile/). I've made a similar demo [tfmobile-mnist-android](https://github.com/nex3z/tfmobile-mnist-android) with batch normalization.

**Comment by longchr123** [comment]
@nex3z  yes, thanks
